cmd/gazelle/integration_test.go

```diff
@@ -1253,6 +1253,82 @@ go_repository(
 		}})
 }
 
+func TestImportReposFromDepToMacro(t *testing.T) {
+	files := []testtools.FileSpec{
+		{Path: "WORKSPACE"},
+		{
+			Path: "repositories.bzl",
+			Content: `
+load("@bazel_gazelle//:deps.bzl", "go_repository")
+
+def go_repositories():
+    go_repository(name = "org_golang_x_net", importpath = "golang.org/x/net", tag = "1.2",)
+    # keep
+    go_repository(name = "org_golang_x_sys", importpath = "golang.org/x/sys", remote = "https://github.com/golang/sys",)
+`,
+		}, {
+			Path: "Gopkg.lock",
+			Content: `# This file is autogenerated, do not edit; changes may be undone by the next 'dep ensure'.
+
+
+[[projects]]
+  name = "github.com/pkg/errors"
+  packages = ["."]
+  revision = "645ef00459ed84a119197bfb8d8205042c6df63d"
+  version = "v0.8.0"
+
+[[projects]]
+  branch = "master"
+  name = "golang.org/x/net"
+  packages = ["context"]
+  revision = "66aacef3dd8a676686c7ae3716979581e8b03c47"
+
+[[projects]]
+  branch = "master"
+  name = "golang.org/x/sys"
+  packages = ["unix"]
+  revision = "bb24a47a89eac6c1227fbcb2ae37a8b9ed323366"
+
+[[projects]]
+  branch = "v2"
+  name = "github.com/go-yaml/yaml"
+  packages = ["."]
+  revision = "cd8b52f8269e0feb286dfeef29f8fe4d5b397e0b"
+
+[solve-meta]
+  analyzer-name = "dep"
+  analyzer-version = 1
+  inputs-digest = "05c1cd69be2c917c0cc4b32942830c2acfa044d8200fdc94716aae48a8083702"
+  solver-name = "gps-cdcl"
+  solver-version = 1
+`,
+		},
+	}
+	dir, cleanup := testtools.CreateFiles(t, files)
+	defer cleanup()
+
+	args := []string{"update-repos", "-build_file_generation", "off", "-from_file", "Gopkg.lock", "-to_macro", "repositories.bzl%go_repositories"}
+	if err := runGazelle(dir, args); err != nil {
+		t.Fatal(err)
+	}
+
+	testtools.CheckFiles(t, dir, []testtools.FileSpec{
+		{
+			Path: "repositories.bzl",
+			Content: `
+load("@bazel_gazelle//:deps.bzl", "go_repository")
+
+def go_repositories():
+    go_repository(name = "org_golang_x_net", build_file_generation = "off", commit = "66aacef3dd8a676686c7ae3716979581e8b03c47", importpath = "golang.org/x/net")
```

Comment:
I think the output is pretty difficult to read with all of the attributes on one line. In `Rule.sync`, could you set the `ForceMultiLine` flag on rules with more than one attribute?

Reply:
It looks like it is not currently possible to print out the attributes on one line :/. I created a PR to fix this issue here: https://github.com/bazelbuild/buildtools/pull/580

Reply:
Thanks for looking into this. I saw you set the `ForceMultiline` flag. So that's getting ignored? Hmm, that's unfortunate.

Reply:
This change has been merged into buildtools. So we will need to update buildtools within Gazelle sometime soon.